src/persist-client/src/internal/state_diff.rs

```diff
@@ -169,6 +170,26 @@ impl<T: Timestamp + Lattice + Codec64> StateDiff<T> {
         diffs
     }
 
+    pub(crate) fn map_blob_inserts<F: for<'a> FnMut(HollowBlobRef<'a, T>)>(&self, f: &mut F) {
+        for spine_diff in self.spine.iter() {
+            match &spine_diff.val {
+                StateFieldValDiff::Insert(()) => {
+                    f(HollowBlobRef::Batch(&spine_diff.key));
+                }
+                StateFieldValDiff::Update((), ()) => panic!("cannot update spine field"),
```

Comment:
we have [something like it for the "single" fields](https://github.com/MaterializeInc/materialize/blob/72d1c73692314a76ad25352265c7c7f1c85aa4e7/src/persist-client/src/internal/state_diff.rs#L364-L365), but in practice the apply_spine code happens to generalize based on a `BTreeMap<HollowBatch<T>, ()>`. it's insidious to have the second codepath, but it also seems nice to be able to do meaningful things on diffs without actually applying them

could easily make this a no-op with a comment instead of a panic, I was just thinking it would be better to be explicit